errors.go

```diff
@@ -0,0 +1,87 @@
+package telebot
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+type ApiError struct {
+	Code        int
+	Description string
+
+	message string
+}
+
+func (err *ApiError) ʔ() string {
+	return err.Description
+}
+
+func badRequest(description ...string) *ApiError {
+	if len(description) == 1 {
+		return &ApiError{400, description[0], ""}
+	}
+
+	return &ApiError{400, description[0], description[1]}
+}
+
+func (err *ApiError) Error() string {
+	canonical := err.message
+	if canonical == "" {
+		butchered := strings.Split(err.Description, ": ")
+		if len(butchered) == 2 {
+			canonical = butchered[1]
+		} else {
+			canonical = err.Description
+		}
+	}
+	return fmt.Sprintf("api error: %s (%d)", canonical, err.Code)
+}
+
+var (
+	//RegExp
+	apiErrorRx = regexp.MustCompile(`{.+"error_code":(\d+),"description":"(.+)"}`)
+
+	//bot creation errors
+	ErrUnauthorized = &ApiError{401, "Unauthorized", ""}
+
+	// not found etc
+	ErrToForwardNotFound = badRequest("Bad Request: message to forward not found")
+	ErrToReplyNotFound   = badRequest("Bad Request: reply message not found")
+	ErrMessageTooLong    = badRequest("Bad Request: message is too long")
+	ErrBlockedByUsr      = &ApiError{401, "Forbidden: bot was blocked by the user", ""}
```

Comment:
Just out of styke, I would push this to the top of the group, so it doesn't break a neat sequence of `badRequest()s`